backend/controllers/questions.js

```diff
@@ -1,29 +1,125 @@
-const Question = require("../models/Question");
-
-const createQuestion = async (req, res) => {
-    res.send("Created Question");
-}
-
-const getQuestionInfo = async (req, res) => {
-    res.send("Question Info");
-}
-
-const updateQuestionInfo = async (req, res) => {
-    res.send("Question Updated");
-}
-
-const deleteQuestion = async (req, res) => {
-    res.send("Question Deleted");
-}
-
-const getAllQuestions = async (req, res) => {
-    res.send("All Questions");
-}
-
-module.exports = {
-    createQuestion,
-    getQuestionInfo,
-    updateQuestionInfo,
-    deleteQuestion,
-    getAllQuestions
+const Question = require("../models/Question");
+const QuestionAnswer = require("../models/QuestionAnswer");
+const SubTopic = require("../models/SubTopic");
+const Topic = require("../models/Topic");
+const { options } = require("../routes/questions");
+
+const createQuestion = async (req, res) => {
+    res.send("Created Question");
+}
+
+const getQuestionInfo = async (req, res) => {
+    res.send("Question Info");
+}
+
+const updateQuestionInfo = async (req, res) => {
+    res.send("Question Updated");
+}
+
+const deleteQuestion = async (req, res) => {
+    res.send("Question Deleted");
+}
+
+const getAllQuestionsById = async (req, res) => {
+    const questions = await Question.findAll({
+        where: {
+            SubTopicId: req.params.id,
+        },
+        include: {
+            model: QuestionAnswer
+        }
+    });
+    const finalQuestions = [];
+
+    for (let index = 0; index < questions.length; index++) {
+        let options = [];
+        let answer = "";
+
+        for (let index2 = 0; index2 < questions[index].QuestionAnswers.length; index2++) {
+            if (questions[index].QuestionAnswers[index2].isCorrect) {
+                answer = questions[index].QuestionAnswers[index2].answerContent;
+            }
+            options.push(questions[index].QuestionAnswers[index2].answerContent);

```

Comment:
You use array functions instead of for loops. E.g some, find, findIndex. The code will be cleaner and most deffinately perform better.